### PR TITLE
Refactor to not use hab-director

### PR DESCRIPTION
**Pull request by jtimberman**
The habitat project removed the hab-director component from the
repository recently. As a result, that component is considered
deprecated and shouldn't be used. This will change the way that private
Depots should be set up, and this commit refactors the cookbook so that
the director is not used.

This has several major changes to the structure of the cookbook.

First, and most obviously, all the `hab-director` resources are removed.

The `install` and `service` recipes are removed and their
content moved to the `default` recipe. There's little value in
decoupling these as the "install" parts are required for all the running
services anyway. It also allows us to reliably subscribe the
hab-builder-api and hab-builder-sessionsrv services to their config file
changes.

Each of the services to run the depot use the `hab_service` resource.
This requires the 0.3.0 version of the `habitat` cookbook for the
`exec_start_options` property. These options are required to set the
listen parameters since we're not using the director that managed those
for us.

The attribute default values are modified. The oauth secrets are
set to nil, and a warning message is logged if they're nil. End-users of
this cookbook will need to populate those values from a custom cookbook
to enable sign-in for the Depot, though that is out of scope for this
cookbook. The fqdn attribute will come from the node's fqdn, and fall
back to its name if the fqdn attribute is not set for some reason. This
can be overridden with normal attribute precedence from a role,
environment, etc.

A test cookbook was created. This will install `jq`, which is used for
parsing output from the config service in the post-convergence inspec
tests.

The .kitchen.yml now allows a user to set environment variables
containing the OAuth application client credentials on their local
system for testing purposes. The README describes this in more detail.

The inspec post-convergence tests are refactored to only test the
outcome - that the services are up and functional, and have the expected
configuration based on the defaults.

Signed-off-by: Joshua Timberman <joshua@chef.io>

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
 version '0.2.0'
```
Comment (bdangit) [inline metadata.rb:7]: do a major bump here as the changes here are no longer backward compatible.

File: README.md [hunk 2]
```diff
@@ -1,17 +1,15 @@
 Habitat Depot
 =============
 
-> Use at your own risk!
```
Comment (bdangit) [inline README.md:4]: Keep this or rewrite as a mentions that Private Depots is not a formal product yet and there's no declared support from Chef.

File: README.md [hunk 3]
```diff
@@ -33,102 +31,46 @@ A service will be created as `hab-depot`.
...
+7. Create packages using that origin, and then upload them to the depot!
+```
+$ hab pkg upload --url http://192.168.96.31/v1/depot somepackage.hart -z $(cat ~/.github/kitchen-habitat-depot)
 ```
 
 ## Credits
 
-@jtimberman's private depot guide
-https://gist.github.com/jtimberman/f939e9c822c581bc7168026f3fa4211c
+Based in part on @jtimberman's [private depot guide](https://gist.github.com/jtimberman/f939e9c822c581bc7168026f3fa4211c)
```
Comment (bdangit) [inline README.md]: out-of-date now.

File: README.md (hunk 2 above)
Comment (jtimberman) [reply README.md:4]: I updated the wording here in e475df3.

**Comment by bdangit** [comment]
👍 thanks @jtimberman!

**Pull request merged**